### PR TITLE
SOC-6229 Fix ignoring system comment with disabled activity type

### PR DESCRIPTION
**Pull request by boubaker**
The check on comment activity type wasn't made before.
This will fix the check on activity type as documented and asked in Feature of SOC-5487

### CODE REVIEW

File: component/core/src/main/java/org/exoplatform/social/core/manager/ActivityManagerImpl.java [hunk 1]
```diff
@@ -452,19 +452,26 @@ public void deleteActivity(String activityId) {
...
+    String commentActivityType = newComment.getType();
     String commentId = newComment.getId();
-    //Activity Type is disable , comment's can't be added
-    //existingActivity.getId() == null for the new activity if it's disabled
-    //comment should be added for the old created activity if it's disabled
-    if (existingActivity!= null && existingActivity.getId() == null && activityType != null && activityTypesRegistry.get(activityType) != null && !activityTypesRegistry.get(activityType)) {
+    // If activity Type is disabled, comment's can't be added
+    // If comment activity Type is disabled, comment's can't be added
+    // If existingActivity.getId() == null for the new activity if it's disabled
+    // comment should be added for the old created activity if it's disabled
```
Comment (thomasdelhomenie) [inline component/core/src/main/java/org/exoplatform/social/core/manager/ActivityManagerImpl.java]: From this comment I understand that comments must be added to old created activities, even if it is now disabled, which is not true anymore with this change, right ?

File: component/core/src/main/java/org/exoplatform/social/core/manager/ActivityManagerImpl.java [hunk 2]
```diff
@@ -452,19 +452,26 @@ public void deleteActivity(String activityId) {
...
+    boolean commentActivityTypeDisabled = commentActivityType != null && activityTypesRegistry.containsKey(commentActivityType) && activityTypesRegistry.get(commentActivityType) == Boolean.FALSE;
+    boolean activityTypeDisabled = activityType != null && activityTypesRegistry.containsKey(activityType) && activityTypesRegistry.get(activityType) == Boolean.FALSE;
```
Comment (thomasdelhomenie) [inline component/core/src/main/java/org/exoplatform/social/core/manager/ActivityManagerImpl.java]: Is `activityTypesRegistry.get(activityType) == Boolean.FALSE` different than `!activityTypesRegistry.get(activityType)` ?

File: component/core/src/main/java/org/exoplatform/social/core/manager/ActivityManagerImpl.java [hunk 3]
```diff
@@ -452,19 +452,26 @@ public void deleteActivity(String activityId) {
...
+    if (commentActivityTypeDisabled || activityTypeDisabled) {
       if (LOG.isDebugEnabled()) {
-        LOG.debug("Comment could not be saved. Activity Type {} has been disabled.", activityType);
+        LOG.debug("Comment could not be saved. Comment activity Type {} has been disabled.", commentActivityType);
```
Comment (thomasdelhomenie) [inline component/core/src/main/java/org/exoplatform/social/core/manager/ActivityManagerImpl.java]: Shouldn't it be "... Activity type or comment activity type {} hes been disabled." ?

File: component/core/src/main/java/org/exoplatform/social/core/manager/ActivityManagerImpl.java (hunk 2 above)
Comment (boubaker) [reply component/core/src/main/java/org/exoplatform/social/core/manager/ActivityManagerImpl.java]: yep, it's different, `null != Boolean.False` doesn't throw an NPE ;)

File: component/core/src/main/java/org/exoplatform/social/core/manager/ActivityManagerImpl.java (hunk 1 above)
Comment (boubaker) [reply component/core/src/main/java/org/exoplatform/social/core/manager/ActivityManagerImpl.java]: What I understood from comment and from issues is that comment shouldn't be added if **comment activity typ**e is disabled. (which wasn't the case switch algorithm)
I invite you to understand that "activity type" and "comment activity type" are two different things.
In addition, if you loog at Test here: https://github.com/exodev/social/blob/0693987e131702b468e4742f249f1e29c4d0f10a/component/core/src/test/java/org/exoplatform/social/core/manager/ActivityManagerTest.java#L1717
You will see that this assumes that the comment is not added.

File: component/core/src/main/java/org/exoplatform/social/core/manager/ActivityManagerImpl.java (hunk 3 above)
Comment (boubaker) [reply component/core/src/main/java/org/exoplatform/social/core/manager/ActivityManagerImpl.java]: modified

File: component/core/src/main/java/org/exoplatform/social/core/manager/ActivityManagerImpl.java (hunk 1 above)
Comment (boubaker) [reply component/core/src/main/java/org/exoplatform/social/core/manager/ActivityManagerImpl.java]: I 'm pretty ok to let people comment on disabled activities, as you asked by Patrice here:
https://jira.exoplatform.org/browse/SOC-5582?focusedCommentId=415198&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-415198

But, the Test case doesn't allow that, so I'm wondering if there was a modification of expected behavior.
@mzorai , you help is needed here to identify what to do in this situation

Comment (mzorai) [reply component/core/src/main/java/org/exoplatform/social/core/manager/ActivityManagerImpl.java]: If comments activity type is disabled ==> no comments added
If an activity type is disabled ==> No comments added to that activity types to not ascend in the AS

Comment (boubaker) [reply component/core/src/main/java/org/exoplatform/social/core/manager/ActivityManagerImpl.java]: ok, thanks @mzorai , @thomasdelhomenie I let you continue the review.

**Comment by boubaker** [comment]
@thomasdelhomenie , the build failed because the activity title was empty in a Unit Test (Fixed by commit https://github.com/exodev/social/pull/597/commits/f81dd726d01e839ad04ed1b4f19e0eed9585d01c)

Could you review again please.

**Comment by thomasdelhomenie** [comment]
Done

**Pull request merged**